pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1494,6 +1494,30 @@ public CompletableFuture<Boolean> checkNonPersistentNonPartitionedTopicExists(St
                                     || actEx instanceof PulsarClientException.TopicDoesNotExistException
                                     || actEx instanceof PulsarAdminException.NotFoundException) {
                                 return CompletableFuture.completedFuture(false);
+                            } else if (actEx instanceof PulsarClientException.NotSupportedException){
+                                /**
+                                 * Summary: For compatibility of
+                                 * {@link org.apache.pulsar.client.impl.BinaryProtoLookupService
+                                 *        #getPartitionedTopicMetadata(TopicName, boolean)}.
+                                 *
+                                 * Explanation:
+                                 * 1. Reason of why getting the error here.
+                                 *   The feature method above was supported at "3.0.6" and "3.3.1", before that the API
+                                 *   "getPartitionedTopicMetadata" will trigger a creation for partitioned topic
+                                 *   metadata automatically even if you just want query it. So the brokers whose version
+                                 *   is less than "3.0.6" and "3.3.1" do not support the new API.
+                                 * 2. The conditions to lead this error occur.
+                                 *   There are 2 brokers in a cluster, and the version is less than "3.0.1", rolling
+                                 *   upgrade brokers to "3.0.6". After the first broker restarted, there is one broker
+                                 *   with version "3.0.6" and another is "3.0.1", and when the internal client tries
+                                 *   to call "getPartitionedTopicMetadata" to the broker with lower version, it will
+                                 *   get this error.
+                                 * 3. Compatibility
+                                 *   Rollback to the original behavior before the fix #22838. Without the fix #22838,
+                                 *   there is an issue that may cause a non-partitioned non-persistent topic and
+                                 *   a partitioned non-persistent topic with the same name to exist at the same time.
+                                 */
```

Comment:
Do we need to pint warn logs(with some rate-limiting or de-duping logs, if too much) here?

Reply:
Printed a warning log here(it will not be too much).